pkg/services/app/add_test.go

```diff
@@ -0,0 +1,209 @@
+		GitProviders: gitProviders,
+	}
+
+	appSrv = app.New(deps)
+
+	defaultParams = app.AddParams{
+		Url:            "https://github.com/foo/bar",
+		Path:           "./kustomize",
+		Branch:         "main",
+		Dir:            ".",
+		DeploymentType: "kustomize",
+		Namespace:      "wego-system",
+	}
+})
+
+var _ = Describe("Add", func() {
+	It("checks for cluster status", func() {
+		err := appSrv.Add(defaultParams)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Expect(kubeClient.GetClusterStatusCallCount()).To(Equal(1))
+	})
+
+	It("gets the cluster name", func() {
+		err := appSrv.Add(defaultParams)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Expect(kubeClient.GetClusterNameCallCount()).To(Equal(1))
+	})
+
+	It("creates and deploys a git secret", func() {
+		fluxClient.CreateSecretGitStub = func(s1, s2, s3 string) ([]byte, error) {
+			return []byte("deploy key"), nil
+		}
+
+		err := appSrv.Add(defaultParams)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Expect(fluxClient.CreateSecretGitCallCount()).To(Equal(1))
+
+		secretRef, repoUrl, namespace := fluxClient.CreateSecretGitArgsForCall(0)
+		Expect(secretRef).To(Equal("weave-gitops-test-cluster"))
+		Expect(repoUrl).To(Equal("ssh://git@github.com/foo/bar.git"))
+		Expect(namespace).To(Equal("wego-system"))
+
+		owner, repoName, deployKey := gitProviders.UploadDeployKeyArgsForCall(0)
+		Expect(owner).To(Equal("foo"))
+		Expect(repoName).To(Equal("bar"))
+		Expect(deployKey).To(Equal([]byte("deploy key")))
+	})
+
+	Context("add app with no config repo", func() {
+		BeforeEach(func() {
+			defaultParams.AutomationRepo = ""
+			defaultParams.CommitManifests = false
+		})
+
+		Describe("generates source manifest", func() {
+			It("creates GitRepository when source type is git", func() {
+				defaultParams.SourceType = string(app.SourceTypeGit)
+
+				err := appSrv.Add(defaultParams)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(fluxClient.CreateSourceGitCallCount()).To(Equal(1))
+
+				name, url, branch, secretRef, namespace := fluxClient.CreateSourceGitArgsForCall(0)
+				Expect(name).To(Equal("bar"))
+				Expect(url).To(Equal("ssh://git@github.com/foo/bar.git"))
+				Expect(branch).To(Equal("main"))
+				Expect(secretRef).To(Equal("weave-gitops-test-cluster"))
+				Expect(namespace).To(Equal("wego-system"))
+			})
+
+			It("creates HelmResitory when source type is helm", func() {
+				defaultParams.Url = "https://charts.kube-ops.io"
+				defaultParams.Chart = "loki"
+
+				err := appSrv.Add(defaultParams)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(fluxClient.CreateSourceHelmCallCount()).To(Equal(1))
+
+				name, url, namespace := fluxClient.CreateSourceHelmArgsForCall(0)
+				Expect(name).To(Equal("loki"))
+				Expect(url).To(Equal("https://charts.kube-ops.io"))
+				Expect(namespace).To(Equal("wego-system"))
+			})
+		})
+
+		Describe("generates application goat", func() {
+			It("creates a kustomization if deployment type kustomize", func() {
```

Comment:
how could we verify that the output for CreateKustomization is right? Is there an easy way to do it with `counterfeiter`

Reply:
That verification goes in the Flux package tests. I didn't add yet.